src/lib/seam/access-codes/use-create-access-code.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  useMutation,
+  type UseMutationResult,
+  useQueryClient,
+} from '@tanstack/react-query'
+import type {
+  AccessCode,
+  AccessCodeGetRequest,
+  LockDevice,
+  OngoingAccessCode,
+  SeamError,
+} from 'seamapi'
+
+import { useSeamClient } from 'lib/seam/use-seam-client.js'
+
+export type UseAccessCodeParams = AccessCodeGetRequest | string
+export type UseAccessCodeData = AccessCode | null
+
+export interface UseCreateAccessCodeData {
+  name: string
+  code: string
+}
+
+export function useCreateAccessCode(
+  device: Pick<LockDevice, 'device_id'>
+): UseMutationResult<OngoingAccessCode, SeamError, UseCreateAccessCodeData> {
+  const { client } = useSeamClient()
+  const queryClient = useQueryClient()
+
+  return useMutation({
+    mutationFn: async (data) => {
+      if (client === null) {
+        throw new Error('Missing seam client')
+      }
+
+      return await client.accessCodes.create({
+        device_id: device.device_id,
+        name: data.name,
+        code: data.code,
+      })
+    },
+    onSuccess: () => {
+      void queryClient.invalidateQueries(['access_codes', 'list'])
+    },
+  })
+}
```

Comment:
As in the names of the exported types? 

```
export type UseCreateAccessCodeParams = never
export type UseCreateAccessCodeData = AccessCode
export type UseCreateAccessCodeMutationParams = AccessCodeCreateRequest
```

Reply:
Yes and the general pattern for the params. All of the public hooks need have "the same" api whenever possible. This should match how this one works: https://github.com/seamapi/react/blob/main/src/lib/seam/connect-webviews/use-create-connect-webview.ts

I tried to rewrite it to match, hopefully I did it right.

Reply:
How's it looking here? https://github.com/seamapi/react/pull/271/commits/8c02d987c098cd3816b0ed750fcfd918d27c4768

Instead of 

```ts
export type UseCreateAccessCodeData = AccessCode
```

I've opted for 

```ts
export interface UseCreateAccessCodeMutationParams {
  name: string
  code: string
}
```

though, because `name` / `code` are nullable in the base access code type:

```
interface AccessCodeBase {
    name?: string;
    code: string | null;
...
}
```

Reply:
Updated the types in https://github.com/seamapi/react/pull/271/commits/dc6cfd1e7b4e29841bb46754fea2fb09b28e991a

Reply:
.... and again in https://github.com/seamapi/react/pull/271/commits/7bddc13db477a8957a2edf5b653c522697e309f1

They now match the suggested types exactly:

```ts
export type UseCreateAccessCodeParams = never
export type UseCreateAccessCodeData = AccessCode
export type UseCreateAccessCodeMutationParams = AccessCodeCreateRequest
```